I've tested locally with below configuration:
```
$ ansible --version
ansible 2.8.0.dev0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/tester/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/tester/pmossakx/ansible/lib/ansible
  executable location = /home/tester/pmossakx/ansible/bin/ansible
  python version = 2.7.12 (default, Nov 12 2018, 14:36:49) [GCC 5.4.0 20160609]

$ git log --pretty=oneline --abbrev-commit -3
3ccc19d Add changelog.
60973f8 Add use_config_proxy option.
eaff384 Simplify checks.

$ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.5 LTS
Release:        16.04
Codename:       xenial

$ pip list |grep docker
docker (3.7.0)

$ docker version
Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       e68fc7a
  Built:            Thu Jan 24 10:56:33 2019
  OS/Arch:          linux/amd64
  Experimental:     false

$ cat /etc/environment
http_proxy=http://myproxy.com:8080
ftp_proxy=http://myproxy.com:8080
https_proxy=http://myproxy.com:8080
no_proxy=127.0.0.1

$ env|grep -i proxy
http_proxy=http://myproxy.com:8080
ftp_proxy=http://myproxy.com:8080
https_proxy=http://myproxy.com:8080
no_proxy=127.0.0.1

# cat /root/.docker/config.json
{
 "proxies":
 {
   "default":
   {
     "httpProxy": "http://myproxy.com:8080",
     "httpsProxy": "http://myproxy.com:8080",
     "noProxy": "127.0.0.1"
   }
 }
}

$ cat playbook.yml
---
- hosts: localhost
  tasks:
    - become: yes
      docker_image:
        name: aptupdate:latest
        build:
          path: .
          use_config_proxy: yes
```

and I have below error:
`buildargs.setdefault(k, v)\r\nAttributeError: 'NoneType' object has no attribute 'setdefault'`
```
$ ansible-playbook -i inventory playbook.yml -vvv

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Shared connection to localhost closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/tester/.ansible/tmp/ansible-tmp-1552987450.79-3358314798076/AnsiballZ_docker_image.py\", line 114, in <module>\r\n    _ansiballz_main()\r\n  File \"/home/tester/.ansible/tmp/ansible-tmp-1552987450.79-3358314798076/AnsiballZ_docker_image.py\", line 106, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/home/tester/.ansible/tmp/ansible-tmp-1552987450.79-3358314798076/AnsiballZ_docker_image.py\", line 49, in invoke_module\r\n    imp.load_module('__main__', mod, module, MOD_DESC)\r\n  File \"/tmp/ansible_docker_image_payload_Klr581/__main__.py\", line 916, in <module>\r\n  File \"/tmp/ansible_docker_image_payload_Klr581/__main__.py\", line 911, in main\r\n  File \"/tmp/ansible_docker_image_payload_Klr581/__main__.py\", line 474, in __init__\r\n  File \"/tmp/ansible_docker_image_payload_Klr581/__main__.py\", line 502, in present\r\n  File \"/tmp/ansible_docker_image_payload_Klr581/__main__.py\", line 725, in build_image\r\n  File \"/usr/local/lib/python2.7/dist-packages/docker/api/build.py\", line 179, in build\r\n    buildargs.setdefault(k, v)\r\nAttributeError: 'NoneType' object has no attribute 'setdefault'\r\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

```

--- a/lib/ansible/modules/cloud/docker/docker_image.py
+++ b/lib/ansible/modules/cloud/docker/docker_image.py
@@ -721,6 +721,10 @@ class ImageManager(DockerBaseClass):
             params['network_mode'] = self.network
         if self.use_config_proxy:
             params['use_config_proxy'] = self.use_config_proxy
+            # Due to a bug in docker-py, it will crash if
+            # use_config_proxy is True and buildargs is None
+            if 'buildargs' not in params:
+                params['buildargs'] = {}
 
         for line in self.client.build(**params):
             # line = json.loads(line)

